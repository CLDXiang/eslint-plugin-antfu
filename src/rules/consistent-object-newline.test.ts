import { expect } from 'vitest'
import { RuleTester } from '../../vendor/rule-tester/src/RuleTester'
import rule, { RULE_NAME } from './consistent-object-newline'

const valids = [
  'const a = { foo: "bar", bar: 2 }',
  'const a = {\nfoo: "bar",\nbar: 2\n}',
  'const a = [1, 2, 3]',
  'const a = [\n1,\n2,\n3\n]',
  'import { foo, bar } from "foo"',
  'import {\nfoo,\nbar\n} from "foo"',
  'const a = [`\n\n`, `\n\n`]',
]

// Check snapshot for fixed code
const invalid = [
  'const a = {\nfoo: "bar", bar: 2 }',
  'const a = {foo: "bar", \nbar: 2\n}',
  'const a = [\n1, 2, 3]',
  'const a = [1, \n2, 3\n]',
  'import {\nfoo, bar } from "foo"',
  'import { foo, \nbar } from "foo"',
  'const a = {foo: "bar", \r\nbar: 2\r\n}',
] as const

const ruleTester: RuleTester = new RuleTester({
  parser: require.resolve('@typescript-eslint/parser'),
})

ruleTester.run(RULE_NAME, rule as any, {
  valid: valids,
  invalid: invalid.map(i => ({
    code: i,
    errors: null,
    onOutput: (output: string) => {
      expect(output).toMatchSnapshot()
    },
  })),
})
