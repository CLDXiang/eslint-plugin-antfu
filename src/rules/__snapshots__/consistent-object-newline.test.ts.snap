// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`consistent-object-newline > invalid > const a = [
1, 2, 3] 1`] = `
"const a = [
1, 
2, 
3
]"
`;

exports[`consistent-object-newline > invalid > const a = [1, 
2, 3
] 1`] = `"const a = [1, 2, 3]"`;

exports[`consistent-object-newline > invalid > const a = {
foo: "bar", bar: 2 } 1`] = `
"const a = {
foo: \\"bar\\", 
bar: 2
 }"
`;

exports[`consistent-object-newline > invalid > const a = {foo: "bar", 
bar: 2
} 1`] = `"const a = {foo: \\"bar\\", bar: 2}"`;

exports[`consistent-object-newline > invalid > const a = {foo: "bar", \\u000d
bar: 2\\u000d
} 1`] = `"const a = {foo: \\"bar\\", bar: 2}"`;

exports[`consistent-object-newline > invalid > import {
foo, bar } from "foo" 1`] = `
"import {
foo, 
bar
 } from \\"foo\\""
`;

exports[`consistent-object-newline > invalid > import { foo, 
bar } from "foo" 1`] = `"import { foo, bar } from \\"foo\\""`;
